components/abtesting/optimizelyXExperiment/src/optimizely-x.js

```diff
@@ -19,12 +24,48 @@ const waitUntil = (truthyFn, callback, delay = 100, interval = 100) => {
   }, interval)
 }
 
-const getOptmizely = () =>
+const getOptimizely = () =>
   window && window.optimizely && window.optimizely.get && window.optimizely
 
-let optimizelyPromise
+/**
+ * Register handler to optimizely ONCE
+ * @param {Object} sdk OptimizelySdk
+ */
+const registerExprimentsEvents = sdk => {
+  if (!wasExprimentsListenerAdded) {
+    sdk.push({
+      type: 'addListener',
+      filter: {type: 'lifecycle', name: 'campaignDecided'},
+      handler: exprimentsActivationHandler
+    })
+    wasExprimentsListenerAdded = true
```

Comment:
Typo here, maybe you wanted to mean `wasExperimentsListenerAdded `.

Reply:
🍋 🍊 exprimentation XD